snap/local/files/run_dcgm_exporter.sh

```diff
@@ -4,11 +4,20 @@ set -euo pipefail
 # Build the argument list for the dcgm-exporter command
 args=()
 
-# Add the dcgm-exporter-address option if it is set. Default: “:9400”
+# Add the dcgm-exporter-address option if it is set.
 dcgm_exporter_address="$(snapctl get dcgm-exporter-address)"
+# Add the dcgm-exporter-metrics-file option if it is set.
+dcgm_exporter_metrics_file_path="$SNAP_COMMON/$(snapctl get dcgm-exporter-metrics-file)"
 
 if [ -n "$dcgm_exporter_address" ]; then
     args+=("-a" "$dcgm_exporter_address")
 fi
 
+# File should be available in the snap data directory under $SNAP_COMMON
+if [[ -f "$dcgm_exporter_metrics_file_path" && -s "$dcgm_exporter_metrics_file_path" ]]; then
+    args+=("-f" "$dcgm_exporter_metrics_file_path")
+else
+    echo "Error: DCGM exporter metrics file not found or empty: $dcgm_exporter_metrics_file_path"
```

Comment:
Thanks, this is better, but still it's confusing to the user, because they see the error, but the exporter still starts and runs successfully.

How does the exporter behave if it's passed a nonexistant file?  Or an empty file?  Can we leave the validation to the exporter itself; as it would be if the user was using this tool without the snap?

Reply:
If the exporter receives an empty file, it will produce no metrics and continue running silently.
If the exporter receives a non-existing file, it will fall back to the default metrics file and continue running silently. 

I think that way, we are a little more explicit to the user without breaking the workflow much.

Reply:
And the justification for not allowing empty files is that if you don't want any metrics, just disable the service

Reply:
Ok that sounds fair, glad it's been thought through :+1:   I think the message needs updating to the effect of "... DCGM exporter is falling back to the default metrics." so it's clear to the user what is happening. :)

Reply:
It's a placeholder for now; I plan to move the validation to the `configure` hook, where we can do the config validation and some service dependency resolution in the next PR. See PR #22 and issue #25. But I will update the message for now, thanks

Reply:
Pushed

Reply:
Got it, thanks